packages/core/src/components/Form.js

```diff
@@ -31,134 +36,42 @@ export default class Form extends Component {
     ) {
-      this.props.onChange(nextState);
+      handleChange(this.props, this.state);
     }
-    this.setState(nextState);
   }
 
-  getStateFromProps(props, inputFormData) {
-    const state = this.state || {};
-    const schema = "schema" in props ? props.schema : this.props.schema;
-    const uiSchema = "uiSchema" in props ? props.uiSchema : this.props.uiSchema;
-    const edit = typeof inputFormData !== "undefined";
-    const liveValidate =
-      "liveValidate" in props ? props.liveValidate : this.props.liveValidate;
-    const mustValidate = edit && !props.noValidate && liveValidate;
-    const rootSchema = schema;
-    const formData = getDefaultFormState(schema, inputFormData, rootSchema);
-    const retrievedSchema = retrieveSchema(schema, rootSchema, formData);
-    const customFormats = props.customFormats;
-    const additionalMetaSchemas = props.additionalMetaSchemas;
-
-    const getCurrentErrors = () => {
-      if (props.noValidate) {
-        return { errors: [], errorSchema: {} };
-      } else if (!props.liveValidate) {
-        return {
-          errors: state.schemaValidationErrors || [],
-          errorSchema: state.schemaValidationErrorSchema || {},
-        };
-      }
-      return {
-        errors: state.errors || [],
-        errorSchema: state.errorSchema || {},
-      };
-    };
-
-    let errors,
-      errorSchema,
-      schemaValidationErrors,
-      schemaValidationErrorSchema;
-    if (mustValidate) {
-      const schemaValidation = this.validate(
-        formData,
-        schema,
-        additionalMetaSchemas,
-        customFormats
-      );
-      errors = schemaValidation.errors;
-      errorSchema = schemaValidation.errorSchema;
-      schemaValidationErrors = errors;
-      schemaValidationErrorSchema = errorSchema;
-    } else {
-      const currentErrors = getCurrentErrors();
-      errors = currentErrors.errors;
-      errorSchema = currentErrors.errorSchema;
-      schemaValidationErrors = state.schemaValidationErrors;
-      schemaValidationErrorSchema = state.schemaValidationErrorSchema;
-    }
-    if (props.extraErrors) {
-      errorSchema = mergeObjects(
-        errorSchema,
-        props.extraErrors,
-        !!"concat arrays"
-      );
-      errors = toErrorList(errorSchema);
-    }
-    const idSchema = toIdSchema(
-      retrievedSchema,
-      uiSchema["ui:rootFieldId"],
-      rootSchema,
-      formData,
-      props.idPrefix
-    );
-    const nextState = {
-      schema,
-      uiSchema,
-      idSchema,
-      formData,
-      edit,
-      errors,
-      errorSchema,
-      additionalMetaSchemas,
-    };
-    if (schemaValidationErrors) {
-      nextState.schemaValidationErrors = schemaValidationErrors;
-      nextState.schemaValidationErrorSchema = schemaValidationErrorSchema;
+  static getDerivedStateFromProps(props, state) {
+    if (
+      Object.keys(props).length !== Object.keys(state.lastProps).length ||
```

Comment:
Please see the discussion here about this change https://github.com/rjsf-team/react-jsonschema-form/pull/2010#discussion_r480266606

Reply:
Hmmm, I'd prefer in this PR if we actually don't change to removing deepEquals and just focus on the refactoring -- we can remove deepEquals in a separate PR. This would make it easier to review that change / make this PR less likely to break things.

Reply:
Sure thing, I'll revert this.